articles/hdinsight/kafka/kafka-troubleshoot-full-disk

```diff
@@ -0,0 +1,83 @@
+---
+title: Broker fails to start due to a full disk in Azure HDInsight
+description: Kafka broker process cannot start since it fails due to disk full error. Any config change will not take into effect since broker process is not starting.
+ms.service: hdinsight
+ms.topic: troubleshooting
+ms.date: 10/08/2021
+---
+
+# Scenario: Brokers are unhealthy or unable to restart due to disk space full issue
+
+This article describes troubleshooting steps and possible resolutions for issues when interacting with Azure HDInsight clusters.
+
+## Issue
+
+Few of the data disks which are used by kafka brokers is full. Kafka broker process cannot start since it fails due to disk full error. Any config change will not take into effect since broker process is not starting.
+
+## Cause
+
+- It's impossible to increase the number of disks or increase the disk size once we have created the kafka cluster.
+- Usually, disk alerts are configured on ambari UI. Whenever the usage of disk increases from 60%, it gives an alert that we need to either scale out or reduce the amount of logs present in kafka cluster.
+- This issue will happen only if you ignore the disk alerts and didn't scale out properly. 
+- Messages are not immediately deleted even though the retention time passes. Messages to be deleted will have a mark of deletion. Then later background cleanup process will delete messages. Also messages in only passive segments are deleted.
+
+> [!IMPORTANT]
+> There are several configs to enhance the log cleaner performance but please be very CAREFUL to apply this enhancement since it may affect producing/consuming messages
+
+## Mitigation
+
+1. Check `server.properties` filesTo find the retention time for all the topics. Sometimes log retention policy is set at topic level. To find retention time configured at topic level, please run the following command:
+
+```
+/usr/hdp/current/kafka-broker/bin/kafka-topics.sh --describe --zookeeper <zookeeper-list> 
+```
+
+
+2. Once we have that information, we need to check for heavy partitions which occupy the maximum space.
+
+```
+# Command to sort the directories based on size:
+du -hs * | sort -rh | head -5 
+```
```

Comment:
```suggestion
```bash
# Command to sort the directories based on size:
du -hs * | sort -rh | head -5 
```
```